src/sentry/api/validators/monitor.py

```diff
@@ -96,27 +101,46 @@ class MonitorValidator(serializers.Serializer):
     type = serializers.ChoiceField(
         choices=zip(MONITOR_TYPES.keys(), MONITOR_TYPES.keys())
     )
+    config = ObjectField()
 
-    def get_default_fields(self):
-        type = self.init_data.get('type', self.object.get('type') if self.object else None)
+    def validate(self, attrs):
+        attrs = super(MonitorValidator, self).validate(attrs)
+        type = self.instance['type'] if self.instance else self.initial_data.get('type')
         if type in MONITOR_TYPES:
             type = MONITOR_TYPES[type]
         if type == MonitorType.CRON_JOB:
-            config = CronJobValidator()
+            validator = CronJobValidator(
+                instance=self.instance.get('config', {}) if self.instance else {},
+                data=attrs.get('config', {}),
+                partial=self.partial,
+            )
+            validator.is_valid(raise_exception=True)
+            attrs['config'] = validator.validated_data
         elif not type:
-            return {}
+            return attrs
         else:
             raise NotImplementedError
-        return {'config': config}
+        return attrs
 
-    def validate_status(self, attrs, source):
-        value = attrs[source]
+    def validate_status(self, value):
         if value:
-            attrs[source] = MONITOR_STATUSES[value]
-        return attrs
+            value = MONITOR_STATUSES[value]
+        return value
 
-    def validate_type(self, attrs, source):
-        value = attrs[source]
+    def validate_type(self, value):
         if value:
-            attrs[source] = MONITOR_TYPES[value]
-        return attrs
+            value = MONITOR_TYPES[value]
+        return value
+
+    def update(self, instance, validated_data):
+        has_config = 'config' in instance or 'config' in validated_data
+        if 'config' in instance or 'config' in validated_data:
```

Comment:
```suggestion
        if has_config:
```

Seems like the same logic here twice.